components/Starknet/modules/architecture_and_concepts/pages/Accounts/deploying_new_accounts.adoc

```diff
@@ -1,61 +1,74 @@
+. Verifies that the address has funds to pay for the deployment.
+. Executes the constructor with the given arguments.
+. Executes the `+__validate_deploy__+` entry point. For more information, see xref:#validate_deploy[].
+. Charges fees from the new account address.
+. Sets the account's nonce to `1`.
 
-* Decide on the account contract that you want to deploy
-* Compute your would-be account address off-chain
-* Send funds to this address
+.Procedure
 
-Once the address has enough funds to pay for the deployment, you can then send a `deploy_account` transaction.
+. For the account contract that you want to deploy, compute the offchain account address.
+// How do I do that?
+. Send funds to this address.
+// How do I do that?
+. When the address has enough funds to pay for the deployment, send a `DEPLOY_ACCOUNT` transaction.
+// How do I do that?
 
-== Transaction flow
+After the deployment has completed successfully, it is recommended to validate the account deployment.
 
-Upon receiving a `deploy_account` transaction, the sequencer will:
+[#DEPLOY_ACCOUNT_restrictions]
+== `DEPLOY_ACCOUNT` contructor restrictions
 
-* Verify that the address has funds to pay for the deployment
-* Execute the constructor with the given arguments
-* Execute the `+__validate_deploy__+` entry point (See below)
-* Charge fee from the new account address
-* Advance the nonce to 1
+The constructor of the `+DEPLOY_ACCOUNT+` transaction has the following limitations:
 
-== Validate deploy
+* Restricted access to `+sequencer_address+` in the `+get_execution_info+` syscall. The syscall returns zero values for `+sequencer_address+`
+* Restricted access to the following syscalls:
+** `+get_block_hash+` for Cairo contracts
+** `+get_sequencer_address+` for Cairo 0 contracts
 
-Two issues might arise from sending a `deploy_account` transaction without any extra validation:
+[#validate_deploy]
+== Validating a `DEPLOY_ACCOUNT` transaction
 
-* Sequencers having the ability to charge arbitrarily high fees, thus potentially draining user funds from a pre-funded account
-* A bad actor having the ability to carry out a sequencer DOS attack by sending multiple, invalid `deploy_account` transactions. This would result in the sequencer not being compensated for work completed.
+Two potential issues with sending a `DEPLOY_ACCOUNT` transaction without any validation are:
 
+* Sequencers can charge arbitrarily high fees, potentially draining user funds from a pre-funded account.
+* A bad actor can carry out a Denial of Service (DoS) attack on the sequencer by sending multiple, invalid `DEPLOY_ACCOUNT` transactions, which would result in the sequencer not being compensated for work completed.
 
-To prevent the scenario described in the first point, a new optional validation entrypoint is provided: `+__validate_deploy__+`.
+You can prevent both issues by using a validation entrypoint called `+__validate_deploy__+`.
```

Comment:
AFAIK `validate_deploy` is mandatory.

Reply:
@ArielElp Can you confirm that it is mandatory?

Reply:
It is mandatory if you want to deploy the account with a deploy_account transaction. This is not a choice, "you" can't and shouldn't prevent DoS, it's the protocol's job of taking care that there are no open DOS vectors. This potential DoS are the reason we are restricting validate_deploy and the constructor during the execution of a deploy_account transaction.

Replace this sentence (and the limitation description below) with:
The requirement to pass validate_deploy after the constructor execution as a prerequisite for charging fees solves 1, and the limitations on validate execution [link to the new limitation section I suggested above] solves 2.